addons/account_payment/models/payment_transaction.py

```diff
@@ -147,7 +147,7 @@ def _create_payment(self, **extra_create_values):
             'payment_method_line_id': payment_method_line.id,
             'payment_token_id': self.token_id.id,
             'payment_transaction_id': self.id,
-            'ref': self.reference,
+            'ref': f'{self.reference} - {self.partner_id.name} - {self.provider_reference or ""}',
```

Comment:
`self.partner_id.display_name` some partner have no name (like invoice contact).

Reply:
Indeed. Although I don't see how an invoice partner's name could be evaluated here, it wouldn't hurt to consider that.

@hote-odoo Can you look into that in both 16.0 and this PR for master?

Reply:
will do :)